### ISSUE TITLE
💬 RFC: Adjustable column width and count in table view

### ISSUE THREAD

**Issue by ShiranHi** [backstage/backstage#27073]
### 🔖 Need

In some cases, the table view contains too many columns, causing the content in each to be cut off, making it difficult to read and overwhelming to consume.

Here is a screenshot:
<img width="1432" alt="image" src="https://github.com/user-attachments/assets/5ec31ba3-04fe-4dfd-8e9f-fbe6ecfddfd2">

### 🎉 Proposal

- I recommend allocating more space to the Description column

- Providing users with control to adjust the number of visible columns, as [Material Design offers a solution for column management](https://material-table-core.github.io/demos/columns/hide/). This allows users to reduce the number of columns and view more relevant content in each one.

### 👀 Have you spent some time to check if this RFC has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

**Comment by Rugvip** [comment]
The description is also available if you hover over the name, and it is also the `title` of the element on focus. I wonder if it would make sense to drop the description column altogether on this page, since space is pretty limited?

**Comment by jerolimov** [comment]
Sorry, I read the new frontend docs, but didn't work with the new extensions on my own.

This is something which might be possible to change with an extension, or?

Overriding the `CatalogTable` with a  `MyCatalogTable` component that again renders the `CatalogTable` with an additional option `{ columnsButton: true }`? :thinking:

**Comment by jerolimov** [comment]
And the tooltip looks today like this:

![image](https://github.com/user-attachments/assets/72bea7f9-c5d1-47fe-b4cd-b89aa546c2f0)

Wouldn't it be great to push it forward in a direction like this (from GitHub):

![image](https://github.com/user-attachments/assets/b5c99c9b-bcc1-467e-b90b-7df5ee557ef5)

What do you both think? Should we open another RFC for this, or do you think it's worth spending some time on that without an issue?

**Comment by Rugvip** [comment]
@jerolimov yep, love the idea! RFC sounds good with a bit of discussion of what information we'd actually highlight and a rough wireframe. We don't have capacity to pick up the work on this up in the maintainer team however, so I'd say only do it if you plan to work in the implementation of this yourself, although you there might be someone else in the community that would be interested working on it too.
